spec/unit_tests/moab/storage_object_validator_spec.rb

```diff
@@ -0,0 +1,87 @@
+require 'spec_helper'
+RSpec.describe Moab::StorageObjectValidator do
+  let(:druid1) { 'xx000xx0000'}
+  let(:druid_path1) { 'spec/fixtures/bad_root01/bad_moab_storage_trunk/xx/000/xx/0000/xx000xx0000' }
+  let(:storage_obj1) { Moab::StorageObject.new(druid1, druid_path1) }
+  let(:storage_obj_validator1) { described_class.new(storage_obj1) }
+
+  describe '#initialize' do
+    it "sets storage_obj_path" do
+      storage_obj_validator = described_class.new(storage_obj1)
+      expect(storage_obj_validator.storage_obj_path.to_s).to eq druid_path1
+    end
+  end
+  context '#validation_errors' do
+    let(:verification_array) { storage_obj_validator1.validation_errors }
+
+    it 'returns correct data structure' do
+      expect(storage_obj_validator1.validation_errors).to be_kind_of Array
+    end
+
+    it 'has item in the array that is the expected structure' do
+      expect(verification_array[0]).to be_kind_of Hash
+    end
+    context 'under version directory' do
+      it 'has missing items' do
+        # v0001
+        expect(verification_array[0]).to eq(1=>"Missing directory: [\"data\", \"manifests\"] Version: v0001")
```

Comment:
feels to me like it'd be clearer to reference the error code in the result hash by its constant name instead of the underlying int val, e.g. 

```ruby
expect(verification_array[0]).to eq(StorageObjectValidator::MISSING_DIR => "Missing directory: [\"data\", \"manifests\"] Version: v0001")
```